specifications/cloud_assembly.md

```diff
@@ -0,0 +1,398 @@
+# Cloud Assembly Specification, Version 1.0
+
+The key words **MUST**, **MUST NOT**, **REQUIRED**, **SHALL**, **SHALL NOT**, **SHOULD**, **SHOULD NOT**,
+**RECOMMENDED**, **MAY**, and **OPTIONAL** in this document are to be interpreted as described in [RFC 2119] when they
+are spelled out in bold, capital letters (as they are shown here).
+
+## Introduction
+A *Cloud Assembly* is a self-contained document container designed to hold the components of *cloud applications*,
+including all the parts that are needed in order to deploy those to a *cloud* provider. This document is the
+specification of the *Cloud Assembly* format as well as requirements imposed on *Cloud Assemblers* and *Cloud Runtimes*.
+
+### Design Goals
+The design goals for the *Cloud Assembly Specification* are the following:
+* The *Cloud Assembly Specification* is extensible.
+* The *Cloud Assembly Specification* is cloud-agnostic.
+* The *Cloud Assembly Specification* is easy to implement and use.
+* The *Cloud Assembly Specification* supports authenticity and integrity guarantees.
+* A *Cloud Assembly* is self-contained, making deployments reproductible.
+
+## Specification
+A *Cloud Assembly* is a ZIP archive that **SHOULD** conform to the [ISO/IEC 21320-1:2015] *Document Container File*
+standard. Use of the `deflate` compression method is **RECOMMENDED** in order to minimize the size of the resulting
```

Comment:
This seems unnecessarily specific and also out of scope for this spec. For example, the JAR specification doesn't seem to have suffered from not telling people that they should pick a particular compression method.

Reply:
OTOH `.class` files don't necessarily compress particularly well. But yeah, maybe this is going a little overboard.